tycho-spi/src/main/java/org/eclipse/tycho/helper/StatusTool.java

```diff
@@ -141,6 +144,13 @@ public static Throwable findException(IStatus status) {
         return null;
     }
 
+    public static void assertNoProblemOf(int severity, IStatus status, Supplier<String> message) {
```

Comment:
I'm not sure this method really does fit its purpose.

I would suggest to use something like


```suggestion
    public static boolen checkStatus(int severity, IStatus status)
```

if one really wants to save a one liner check.

The caller should the use 

if (StatusTool.checkStatus(....)) {

} else {
  //print log message and/or throw a suitable exception e.g CoreException or MojoFailureException or ...
}

Reply:
As there is already `IStatus.matches(int)` a `checkStatus` helper method doesn't seem to add much value here.
If the generalized method doesn't fit here, lets just return to the specialization defined in `P2RepositoryManager`.